### PR TITLE
feat: Add GH Action Template

### LINKED ISSUE #440: template: Github actions starter action
- A build action, and a release/deployment action (that deploys to a theoretical azure app service)
- Deployment should be reasonably explained/obvious how to scale to multiple environments
- Add this behind a new template parameter (in template.json)
- Don't include dependency caching. In my observed experience this actually makes the build slower. And its just more "stuff" for someone to have to grok.

### PR DESCRIPTION
**Pull request by BenjaminMichaelis**
Fixes #440

### CODE REVIEW

File: templates/Coalesce.Vue.Template/content/.github/workflows/build-test-and-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+# 1. Set up the following secrets in your repository:
+#   AZURE_WEBAPP_PUBLISH_PROFILE
+#
+# 2. Set up the production Environment in the GitHub Repository Settings.
+#   Environment Protection Rules are recommended to prevent accidental deployments.
+#   https://docs.github.com/actions/managing-workflow-runs-and-deployments/managing-deployments/managing-environments-for-deployment#deployment-protection-rules
+#
+# 3. Change these variables for your configuration:
+env:
+  AZURE_WEBAPP_NAME: Coalesce.Starter.Vue # set this to your app service name
```
Comment (ascott18) [inline templates/Coalesce.Vue.Template/content/.github/workflows/build-test-and-deploy.yml:23]: Might put the placeholder value as something that could plausibly be an appservice name (replace dashes with dots). It'll be wrong, but it'll help hint to a confused user that we're looking for the app service name, not the csproj name.

File: templates/Coalesce.Vue.Template/content/Coalesce.Starter.Vue.Web/src/App.vue [hunk 2]
```diff
@@ -108,8 +108,13 @@ import { useTheme } from "vuetify";
 
 const drawer = ref<boolean | null>(null);
 
+//#if Identity
 const router = useRouter();
 const { userInfo } = useUser();
+//#else
+useRouter();
+useUser();
+//#endif
```
Comment (ascott18) [inline templates/Coalesce.Vue.Template/content/Coalesce.Starter.Vue.Web/src/App.vue:117]:
```suggestion
//#endif
```

**Review by ascott18 — CHANGES_REQUESTED** [review]

File: templates/Coalesce.Vue.Template/content/.github/workflows/build-test-and-deploy.yml (hunk 1 above)
Comment (BenjaminMichaelis) [reply templates/Coalesce.Vue.Template/content/.github/workflows/build-test-and-deploy.yml]: Update to `MY_WEBAPP_NAME`

Comment (ascott18) [reply templates/Coalesce.Vue.Template/content/.github/workflows/build-test-and-deploy.yml]: Sorry, I said that backwards. I meant replace dots with dashes

Comment (ascott18) [reply templates/Coalesce.Vue.Template/content/.github/workflows/build-test-and-deploy.yml]: underscores aren't valid in app service names 
image

Comment (BenjaminMichaelis) [reply templates/Coalesce.Vue.Template/content/.github/workflows/build-test-and-deploy.yml]: I figured 😄

Comment (BenjaminMichaelis) [reply templates/Coalesce.Vue.Template/content/.github/workflows/build-test-and-deploy.yml]: Offline convo - keeping as an invalid name to match GH docs: https://docs.github.com/en/actions/use-cases-and-examples/deploying/deploying-net-to-azure-app-service#creating-the-workflow

Correct naming guidelines for reference: https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules?WT.mc_id=8B97120A00B57354#microsoftweb

**Pull request merged**